src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java

```diff
@@ -104,6 +119,29 @@ public void setNamespace(String namespace) {
     this.namespaces = StringUtils.isBlank(namespace)?null:namespace.split(" ");
   }
 
+  // https://wiki.jenkins-ci.org/display/JENKINS/Credentials+Plugin
+  // http://javadoc.jenkins-ci.org/credentials/com/cloudbees/plugins/credentials/common/AbstractIdCredentialsListBoxModel.html
+  // https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java
+  public static ListBoxModel doFillCredentialsIdItems(String credentialsId) {
+    if (credentialsId == null) {
+      credentialsId = "";
+    }
+
+    if (!Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER)) {
+      // Important! Otherwise you expose credentials metadata to random web requests.
+      return new StandardListBoxModel().includeCurrentValue(credentialsId);
+    }
+
+    return new StandardListBoxModel()
+            .includeEmptyValue()
+            .includeAs(ACL.SYSTEM, Jenkins.getInstance(), OpenShiftTokenCredentials.class)
+            //.includeAs(ACL.SYSTEM, Jenkins.getInstance(), StandardUsernamePasswordCredentials.class)
```

Comment:
I'm fine with that too, though we should ensure that we are as consistent as possible between the credential types offered with sync plugin and the credential types offered with the client plugin.